### ISSUE TITLE
config file metadata not saved for AWS::Events::Rule resource

### ISSUE THREAD

**Issue by tazatwell** [aws-cloudformation/cfn-python-lint#1298]
*cfn-lint version: 0.25.0

*Description of issue.*

Here's `sample-template.yaml`

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'
Metadata: # ignore nodeJS 8.10 obsolete warning
  cfn-lint:
    config:
      ignore_checks:
      - W2531
      #- W3005

Resources:
  Lambda:
    Type: 'AWS::Serverless::Function'
    Properties:
      Handler: "index.handler"
      Runtime: "nodejs8.10"

  EventRule:
    DependsOn:
      - Lambda
    Type: AWS::Events::Rule
    Properties:
      EventPattern:
        source:
          - "aws.ec2"
        detail-type:
          - "EC2 Instance State-change Notification"
      Targets:
        - Arn:
            Fn::GetAtt: Lambda.Arn
          Id:
            Ref: Lambda
```

and my `.cfnlintrc`  file:

```
templates:
- sample-template.yaml
ignore_checks:
- W3005
```

and current dir:

```
~/Documents/workspace/temp|
⇒  tree -a .                               
.
├── .cfnlintrc
└── sample-template.yaml

0 directories, 2 files
~/Documents/workspace/temp|
⇒  
```

cfn-lint reports a `W3005` warning even though it is ignored in the `.cfnlintrc` file:
```
~/Documents/workspace/temp|
⇒  cfn-lint          
W3005 Obsolete DependsOn on resource (Lambda), dependency already en
forced by a "Ref" at Resources/EventRule/Properties/Targets/0/Id/Ref
sample-template.yaml:19:9

~/Documents/workspace/temp|
⇒  
```

However, uncommenting the ignore W3005 line in the Metadata for `sample-template.yaml` will correctly ignore the error.

Perhaps metadata is ignored for similar reasons to https://github.com/awslabs/serverless-application-model/issues/264 and #1294, but in this case it is the `.cfnlintrc` file data that is being ignored, not in the resource's metadata.

I understand that this issue may take time to be resolved, I just wanted to bring this to your attention.

Thanks :)

**Comment by tazatwell** [comment]
From `README.md`:


### Precedence

cfn-lint applies the configuration from the CloudFormation Metadata first and then overrides those values with anything specified in the CLI.


Is this issue due to precedence? Does `.cfnlintrc` configurations have the same precedence as values passed through the CLI?

Thanks in advance :)

**Comment by kddejong** [comment]
yea we tried to move from more generic to more specific when doing the precedence.
1. `.cfnlintrc` This can apply to all templates and all runs
2. Template Metadata.  Is done as it always applies to the template
3. CLI parameters are applied.  This is to allow for a specific run to have its own configuration.

All parameters at this point override the higher levels.  We have talked about merging things like ignore and append as the intention isn't always to override.

**Comment by tazatwell** [comment]
Hi, thanks for the clarification. I'll make a quick PR to `README.md` with that information.
